java.util.regex.PatternSyntaxException: Dangling meta character '?' near index 0 ?/.bitcoin-s/(testnet3|mainnet|regtest)/?$

This is occuring on an umbrel build where we combine our oracle server backend with the proxy server frontend via `docker-compose`

This is where this is coming from I believe

https://github.com/bitcoin-s/bitcoin-s/blob/4f7b6422ea3390f276733ee3c0fb21d426bde751/app-commons/src/main/scala/org/bitcoins/commons/config/AppConfig.scala#L251

This is where you can build our docker infra to reproduce. The container stops immediately on startup is my understanding. 

#### Reproducing

Clone this:

https://github.com/Christewart/umbrel/tree/2021-09-14-issue-3666

```
$ cd apps/krystal-bull
$ docker-compose up
```

Should give you this exception: 

![Screenshot from 2021-09-14 14-52-19](https://user-images.githubusercontent.com/3514957/133324599-ed111bbd-e89f-462c-a33c-d2993a059ac6.png)

Stack trace: 

```
 docker-compose up                                                                     ✔  12803  14:51:50
WARNING: The APP_DATA_DIR variable is not set. Defaulting to a blank string.
Starting krystal-bull_oracleserver_1 ... done
Starting krystal-bull_web_1          ... done
Attaching to krystal-bull_oracleserver_1, krystal-bull_web_1
oracleserver_1  | Exception in thread "main" java.lang.ExceptionInInitializerError
oracleserver_1  |       at org.bitcoins.commons.util.DatadirParser.datadirConfig$lzycompute(DatadirParser.scala:25)
oracleserver_1  |       at org.bitcoins.commons.util.DatadirParser.datadirConfig(DatadirParser.scala:23)
oracleserver_1  |       at org.bitcoins.commons.util.DatadirParser.baseConfig$lzycompute(DatadirParser.scala:44)
oracleserver_1  |       at org.bitcoins.commons.util.DatadirParser.baseConfig(DatadirParser.scala:34)
oracleserver_1  |       at org.bitcoins.commons.util.DatadirParser.datadir$lzycompute(DatadirParser.scala:53)
oracleserver_1  |       at org.bitcoins.commons.util.DatadirParser.datadir(DatadirParser.scala:52)
oracleserver_1  |       at org.bitcoins.commons.util.DatadirParser.networkDir(DatadirParser.scala:62)
oracleserver_1  |       at org.bitcoins.oracle.server.OracleServerMain$.delayedEndpoint$org$bitcoins$oracle$server$OracleServerMain$1(OracleServerMain.scala:74)
oracleserver_1  |       at org.bitcoins.oracle.server.OracleServerMain$delayedInit$body.apply(OracleServerMain.scala:61)
oracleserver_1  |       at scala.Function0.apply$mcV$sp(Function0.scala:39)
oracleserver_1  |       at scala.Function0.apply$mcV$sp$(Function0.scala:39)
oracleserver_1  |       at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:17)
oracleserver_1  |       at scala.App.$anonfun$main$1(App.scala:76)
oracleserver_1  |       at scala.App.$anonfun$main$1$adapted(App.scala:76)
oracleserver_1  |       at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:563)
oracleserver_1  |       at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:561)
oracleserver_1  |       at scala.collection.AbstractIterable.foreach(Iterable.scala:919)
oracleserver_1  |       at scala.App.main(App.scala:76)
oracleserver_1  |       at scala.App.main$(App.scala:74)
oracleserver_1  |       at org.bitcoins.oracle.server.OracleServerMain$.main(OracleServerMain.scala:61)
oracleserver_1  |       at org.bitcoins.oracle.server.OracleServerMain.main(OracleServerMain.scala)
oracleserver_1  | Caused by: java.util.regex.PatternSyntaxException: Dangling meta character '?' near index 0
oracleserver_1  | ?/.bitcoin-s/(testnet3|mainnet|regtest)/?$
oracleserver_1  | ^
oracleserver_1  |       at java.base/java.util.regex.Pattern.error(Pattern.java:2028)
oracleserver_1  |       at java.base/java.util.regex.Pattern.sequence(Pattern.java:2203)
oracleserver_1  |       at java.base/java.util.regex.Pattern.expr(Pattern.java:2069)
oracleserver_1  |       at java.base/java.util.regex.Pattern.compile(Pattern.java:1783)
oracleserver_1  |       at java.base/java.util.regex.Pattern.<init>(Pattern.java:1430)
oracleserver_1  |       at java.base/java.util.regex.Pattern.compile(Pattern.java:1069)
oracleserver_1  |       at scala.util.matching.Regex.<init>(Regex.scala:231)
oracleserver_1  |       at scala.collection.StringOps$.r$extension(StringOps.scala:864)
oracleserver_1  |       at org.bitcoins.commons.config.AppConfig$.<clinit>(AppConfig.scala:251)
oracleserver_1  |       ... 21 more
web_1           |
```

--- a/project/CommonSettings.scala
+++ b/project/CommonSettings.scala
@@ -183,6 +183,8 @@ object CommonSettings {
       //the default provided by sbt native packager
       //which is 'demiourgos728'
       Docker / daemonUser := "bitcoin-s",
+      //needed for umbrel to run
+      Docker / daemonUserUid := Some("1000"),
       Docker / packageName := packageName.value,
       Docker / version := version.value,
       dockerUpdateLatest := isSnapshot.value

